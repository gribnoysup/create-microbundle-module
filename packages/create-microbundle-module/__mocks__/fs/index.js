const path = require('path');
// INFO: We will replace actual file system with
// in-memory implementation
const { vol: volume, fs: memfs } = require('memfs');

const fs = require.requireActual('fs');

// INFO: For this to work properly, we will copy
// all the templates to in-memory volume
const templatesDir = (...pathSegments) =>
  path.resolve(__dirname, '..', '..', 'templates', ...pathSegments);

const templateFiles = fs.readdirSync(templatesDir()).reduce((acc, fileName) => {
  acc[fileName] = fs.readFileSync(templatesDir(fileName), 'utf-8');
  return acc;
}, {});

volume.fromJSON(templateFiles, templatesDir());

module.exports = memfs;
