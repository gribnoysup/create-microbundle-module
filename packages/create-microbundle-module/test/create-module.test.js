import path from 'path';
import os from 'os';

import fs from 'fs-extra';
import dirTree from 'directory-tree';

import createModule from '../src/create-module';

describe('createModule', async () => {
  const fixtureModuleDir = path.resolve(os.tmpdir(), 'my-awesome-module');

  beforeAll(async done => {
    await createModule(fixtureModuleDir, {
      target: 'web',
      scriptsVersion: 'create-microbundle-module',
    });

    done();
  });

  it('should create initial files structure', () => {
    const files = [];

    dirTree(
      fixtureModuleDir,
      {
        exclude: /\.git\/.+/,
        normalizePath: true,
      },
      item => {
        files.push(item.path.replace(fixtureModuleDir, 'my-awesome-module'));
      }
    );

    expect(files).toMatchSnapshot();
  });

  it('should create source entry', async () => {
    const entry = await fs.readFile(
      path.join(fixtureModuleDir, 'src', 'my-awesome-module.js'),
      'utf-8'
    );

    expect(entry).toMatchSnapshot();
  });

  it('should create test file', async () => {
    const test = await fs.readFile(
      path.join(fixtureModuleDir, 'test', 'my-awesome-module.test.js'),
      'utf-8'
    );

    expect(test).toMatchSnapshot();
  });

  it('should create .gitignore file', async () => {
    const gitignore = await fs.readFile(
      path.join(fixtureModuleDir, '.gitignore'),
      'utf-8'
    );

    expect(gitignore).toMatchSnapshot();
  });

  it('should create package.json file', async () => {
    const pkg = await fs.readJSON(path.join(fixtureModuleDir, 'package.json'));

    // INFO: package.author is dynamic and depends on git config.
    // To avoid tests failing with unmatched snapshot, we will
    // "delete" author
    delete pkg.author;

    expect(pkg).toMatchSnapshot();
  });

  afterAll(async done => {
    await fs.remove(fixtureModuleDir);

    done();
  });
});
