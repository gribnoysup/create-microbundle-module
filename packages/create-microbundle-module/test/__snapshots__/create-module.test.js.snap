// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createModule should create .gitignore file 1`] = `
"node_modules
dist
*.log
.*
!.gitignore
!.microbundle-module-config
"
`;

exports[`createModule should create initial files structure 1`] = `
Array [
  "my-awesome-module/.gitignore",
  "my-awesome-module/package.json",
  "my-awesome-module/src/my-awesome-module.js",
  "my-awesome-module/test/my-awesome-module.test.js",
]
`;

exports[`createModule should create package.json file 1`] = `
Object {
  "files": Array [
    "dist",
  ],
  "husky": Object {
    "hooks": Object {
      "pre-commit": "npm run lint-staged",
      "pre-push": "npm test",
    },
  },
  "license": "ISC",
  "main": "dist/my-awesome-module.js",
  "module": "dist/my-awesome-module.m.js",
  "name": "my-awesome-module",
  "scripts": Object {
    "build": "microbundle-module-scripts build",
    "lint-staged": "microbundle-module-scripts lint-staged",
    "start": "microbundle-module-scripts start",
    "test": "microbundle-module-scripts test",
  },
  "source": "src/my-awesome-module.js",
  "target": "web",
  "umd:main": "dist/my-awesome-module.umd.js",
  "version": "0.1.0",
}
`;

exports[`createModule should create source entry 1`] = `
"export default function myAwesomeModule() {
  /** You are ready to write your awesome module 🎉 */
}
"
`;

exports[`createModule should create test file 1`] = `
"import myAwesomeModule from '../src/my-awesome-module';

describe('myAwesomeModule', () => {
  /** Don't be shy, write your first test 😉 */
});
"
`;
